addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java

```diff
@@ -242,15 +242,22 @@ public void run() {
                             }
 
                             logger.debug("Receive Message = {}", SerialMessage.bb2hex(buffer));
-                            SerialMessage recvMessage = new SerialMessage(buffer);
-                            if (recvMessage.isValid) {
-                                logger.trace("Message is valid, sending ACK");
-                                sendResponse(ACK);
-
-                                incomingMessage(recvMessage);
-                            } else {
-                                logger.error("Message is invalid, discarding");
-                                sendResponse(NAK);
+                            SerialMessage recvMessage = null;
+                            try {
+                                recvMessage = new SerialMessage(buffer);
+                            } catch (final IllegalArgumentException ex) {
+                                logger.warn("Serial message processing failed: {}", ex.getMessage());
+                            }
+                            if (recvMessage != null) {
```

Comment:
I have used this check to catch only the IllegalArgumentException that is thrown by the constructor of the serial message object and not other ones, too.